nat-lab/tests/uniffi/libtelio_remote.py

```diff
@@ -75,19 +76,21 @@ def next_event(self):
 
 class TelioLoggerCbImpl(libtelio.TelioLoggerCb):
     def __init__(self):
+        self.lock = Lock()
         try:
             os.remove(TCLI_LOG)
         except FileNotFoundError:
             pass
 
     def log(self, log_level, payload):
-        with open(TCLI_LOG, "a", encoding="utf-8") as logfile:
-            try:
-                logfile.write(f"{datetime.datetime.now()} {log_level} {payload}\n")
-            except IOError as e:
-                logfile.write(
-                    f"{datetime.datetime.now()} Failed to write logline due to error {str(e)}\n"
-                )
+        with self.lock:
+            with open(TCLI_LOG, "a", encoding="utf-8") as logfile:
+                try:
+                    logfile.write(f"{datetime.datetime.now()} {log_level} {payload}\n")
+                except IOError as e:
+                    logfile.write(
+                        f"{datetime.datetime.now()} Failed to write logline due to error {str(e)}\n"
+                    )
```

Comment:
nit: Maybe we should switch to [logging](https://docs.python.org/3/library/logging.html#thread-safety) which claims to be thread safe by design?

Reply:
I don't see big wins here from using it.